handlers/delete_namespaces.go

```diff
@@ -0,0 +1,49 @@
+package handlers
+
+import (
+	"fmt"
+
+	"github.com/deis/k8s-claimer/k8s"
+	k8sapi "k8s.io/kubernetes/pkg/api"
+	labels "k8s.io/kubernetes/pkg/labels"
+)
+
+type errListNamespaces struct {
+	origErr error
+}
+
+func (e errListNamespaces) Error() string {
+	return fmt.Sprintf("error listing namespaces (%s)", e.origErr)
+}
+
+type errDeleteNamespaces struct {
+	origErrs []error
+}
+
+func (e errDeleteNamespaces) Error() string {
+	return fmt.Sprintf("error deleting namespaces (%+v)", e.origErrs)
+}
+
+// deleteNamespaces deletes all namespaces listed under all labels in namespaces.List, except for
+// the namespaces in skip, "kube-system" and "default"
+func deleteNamespaces(namespaces k8s.NamespaceListerDeleter, skip map[string]struct{}) error {
+	namespacesList, err := namespaces.List(k8sapi.ListOptions{LabelSelector: labels.Everything()})
+	if err != nil {
+		return errListNamespaces{origErr: err}
+	}
+	// TODO: delete concurrently https://github.com/deis/k8s-claimer/issues/49
+	var errs []error
+	for _, namespace := range namespacesList.Items {
+		_, inSkip := skip[namespace.Name]
+		isDefault := namespace.Name == "kube-system" || namespace.Name == "default"
```

Comment:
Yea, this can use that map. It's defined globally to the `handlers` package, so that should be good enough.

Fixed in 67ce79f0f7a4406e12df4870b768d09e01372829